src/Compiler/Checking/Expressions/CheckArrayOrListComputedExpressions.fs

```diff
@@ -68,66 +150,74 @@ let TcArrayOrListComputedExpression (cenv: TcFileState) env (overallTy: OverallT
                 errorR (Deprecated(FSComp.SR.tcExpressionWithIfRequiresParenthesis (), m))
             | _ -> ()
 
-            let replacementExpr =
-                if isArray then
+            if
+                containsRangeExpressions elems
+                && g.langVersion.SupportsFeature LanguageFeature.AllowMixedRangesAndValuesInSeqExpressions
+            then
+                TcMixedSequencesWithRanges cenv env overallTy tpenv isArray elems m
+            else
+                if containsRangeExpressions elems then
```

Comment:
We should probably call `containsRangeExpressions` once and store the result before this `if`, otherwise this will traverse the full elements list twice until this feature is out of preview.